pgjdbc/src/main/java/org/postgresql/core/ParameterContext.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2020, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.core;
+
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ParameterContext {
+  public static final ParameterContext EMPTY_CONTEXT = new ParameterContext();
+  private static final List<Integer> NO_PLACEHOLDERS = new ArrayList<>();
```

Comment:
Why is this needed? Couldn't Collections.emptyList() be used?

https://docs.oracle.com/javase/8/docs/api/java/util/Collections.html#emptyList--

Reply:
I was not familiar with that construct. Looks like a good idea.